lib/v2/iqiyi/video.js

```diff
@@ -1,43 +1,45 @@
-const got = require('@/utils/got');
 const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const logger = require('@/utils/logger');
 // /iqiyi/user/video/:uid
 // http://localhost:1200/iqiyi/user/video/2289191062
 module.exports = async (ctx) => {
     const uid = ctx.params.uid;
+    const link = `https://www.iqiyi.com/u/${uid}/videos`;
 
-    const response = await got({
-        method: 'get',
-        url: `http://www.iqiyi.com/u/${uid}/v`,
-        headers: {
-            Host: 'www.iqiyi.com',
-            Referer: `http://www.iqiyi.com/u/${uid}/v`,
-        },
+    // Use puppeteer because iqiyi page has a delay.
+    const browser = await require('@/utils/puppeteer')();
+    const data = await ctx.cache.tryGet(link, async () => {
+        const page = await browser.newPage();
+        await page.setRequestInterception(true);
+        page.on('request', (request) => {
+            request.resourceType() === 'document' || request.resourceType() === 'script' ? request.continue() : request.abort();
+        });
+        logger.debug(`Requesting ${link}`);
+        await page.goto(link, {
+            waitUntil: 'domcontentloaded',
+        });
+        await page.waitForSelector('li.pic-txt-li');
+        return await page.content();
```

Comment:
Consider storing `ctx.state.data.item` instead of raw HTML data which uses the cache more efficiently